### PR TITLE
Prioritise consultations in the policy and engagement section

### PR DESCRIPTION
**Pull request by leenagupte**
Trello: https://trello.com/c/lZhCKzi7

### What's changed
Promote consultation content in the Policy and engagement section, and render them in a highlight box.

Example topic page: https://govuk-collections-pr-625.herokuapp.com/education/funding-and-finance-for-students

### Before
screen shot 2018-04-18 at 18 08 33

### After
screen shot 2018-04-18 at 18 09 26

### CODE REVIEW

File: app/presenters/supergroups/policy_and_engagement.rb [hunk 1]
```diff
@@ -8,6 +8,15 @@ def initialize
 
     def tagged_content(taxon_id)
       @content = MostRecentContent.fetch(content_id: taxon_id, filter_content_purpose_supergroup: @name)
+      @content.select do |content_item|
+        consultation?(content_item.content_store_document_type)
+      end | @content
```
Comment (cbaines) [inline app/presenters/supergroups/policy_and_engagement.rb]: `|` is bitwise or in Ruby right?

File: app/presenters/supergroups/supergroup.rb [hunk 2]
```diff
@@ -53,5 +53,9 @@ def document_list(taxon_id)
         data
       end
     end
+
+    def special_format_count(*)
+      0
```
Comment (vanitabarrett) [inline app/presenters/supergroups/supergroup.rb]: Can we set '3' as the default, so we don't have to override this method in every supergroup class?

File: test/presenters/supergroups/policy_and_engagement_test.rb [hunk 3]
```diff
@@ -12,21 +12,177 @@
...
+
+        MostRecentContent.any_instance
+          .stubs(:fetch)
+          .returns(tagged_content(tagged_document_list))
+
+        assert_equal expected_results(expected_order), policy_and_engagement_supergroup.document_list(taxon_id)
+      end
+
+      describe '#special_format_count' do
+        it 'only show applies special formatting to consultations' do
```
Comment (vanitabarrett) [inline test/presenters/supergroups/policy_and_engagement_test.rb]: Not sure what you're trying to say in this test name... (or the one underneath)

File: test/presenters/supergroups/policy_and_engagement_test.rb [hunk 4]
```diff
@@ -12,21 +12,177 @@
...
+      describe '#consultation_closing_date' do
+        it 'gets the closing date of past consultations' do
+          MostRecentContent.any_instance
+            .stubs(:fetch)
+            .returns(section_tagged_content_list('open_consultation'))
+
+          assert_equal expected_result('open_consultation'), policy_and_engagement_supergroup.document_list(taxon_id)
+        end
+
+        it 'gets the closing date of future consultations' do
```
Comment (vanitabarrett) [inline test/presenters/supergroups/policy_and_engagement_test.rb]: I'm a bit confused about the difference between this test and `gets the closing date of past consultations`. We seem to be testing both with a document type of `open_consultation`, but the test names suggest we're testing past (closed consultations) and future (open consultations)?

Comment (leenagupte) [reply test/presenters/supergroups/policy_and_engagement_test.rb]: @vanitabarrett The naming is bad. The description of the consultation closing date changes depending on whether it's in the past or future.

e.g. "Closing date 10 July 2018" vs "Date closed 10 July 2017"

In `gets the closing date of past consultations` the closing date is in 2017, and in `gets the closing date of future consultations` the date is in 2018.

File: app/presenters/supergroups/policy_and_engagement.rb (hunk 1 above)
Comment (leenagupte) [reply app/presenters/supergroups/policy_and_engagement.rb]: Yeah. I was trying to find a way to get consultations from the list of documents without losing the other documents. There's probably a nicer way to do this.

File: test/presenters/supergroups/policy_and_engagement_test.rb (hunk 3 above)
Comment (leenagupte) [reply test/presenters/supergroups/policy_and_engagement_test.rb]: @vanitabarrett I'll rename this! That's some bad English right there!

**Comment by leenagupte** [comment]
@vanitabarrett  I've replaced by #626 and addressed your comments there. On the request of @miaallers, I've also removed the "featured" flag from the highlight boxes in the new PR.

**Pull request closed without merging**